### PR TITLE
Defect/626 state input

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: webapp/src/main/webapp/src/app/admin/tenant/component/property-override-tree-table/property-override-tree-table.component.ts [hunk 1]
```diff
@@ -157,6 +157,18 @@ export class PropertyOverrideTreeTableComponent
...
+  readonlyValue(property: ConfigurationProperty): any {
+    const value = this.formGroup.value[property.key];
+    if (value != null) {
+      return value;
+    }
+    const defaultValue = this.defaults[property.key];
+    if (defaultValue != null) {
+      return defaultValue;
+    }
+    return '';
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/component/property-override-tree-table/property-override-tree-table.component.ts]: unrelated fix for rendering readonly values. @ftdholbrook i'm curious if we set password for tenant will in come back as **** in the readonly view? my guess is it will but i wasn't able to confirm locally

File: webapp/src/main/webapp/src/app/admin/tenant/component/tenant-form/tenant-form.component.html [hunk 2]
```diff
@@ -275,6 +225,56 @@ <h1 class="h2" heading>
       </div>
     </ng-container>
 
+    <div
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/component/tenant-form/tenant-form.component.html]: moved tenant and data set form fields above the name field since the default name will change when you change the base tenant

File: webapp/src/main/webapp/src/app/admin/tenant/component/tenant-form/tenant-form.component.ts [hunk 3]
```diff
@@ -251,11 +252,10 @@ function setDefaultDatabaseNameAndUsername(
     }, [])
     // for every datasource name apply defaults
     .forEach(source => {
-      const usernameControl =
-        formGroup.controls[`datasources.${source}.username`];
-      if (usernameControl.pristine) {
-        usernameControl.patchValue(defaultUsername);
-      }
+      patch(
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/component/tenant-form/tenant-form.component.ts]: when you change tenant now the sandbox label and config props will correctly change to match that tenants overrides

File: webapp/src/main/webapp/src/app/admin/tenant/pages/tenant/tenant.component.ts [hunk 4]
```diff
@@ -54,8 +60,14 @@ export class TenantComponent implements OnDestroy {
...
     private translationLoader: RdwTranslateLoader
   ) {
-    const sandboxDataSets$ = this.service.getSandboxDataSets().pipe(share());
-    const tenants$ = this.service.getAll('TENANT').pipe(share());
+    const sandboxDataSets$ = this.service.getSandboxDataSets().pipe(
+      map(values => values.slice().sort(byLabel)),
+      share()
+    );
+    const tenants$ = this.service.getAll('TENANT').pipe(
+      map(values => values.slice().sort(byLabel)),
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/pages/tenant/tenant.component.ts]: tenants and data sets are now sorted by label on the sandbox form

File: webapp/src/main/webapp/src/app/shared/support/support.ts [hunk 5]
```diff
@@ -553,7 +553,18 @@ export function rightDifference(a: any, b: any): any {
  */
 export function valued(value: any): any {
   return Object.entries(value).reduce((valued, [key, value]) => {
-    if (value != null && !(typeof value === 'string' && value.length === 0)) {
+    if (
+      // null or undefined
+      value != null &&
+      // empty strings
+      !(typeof value === 'string' && value.length === 0) &&
+      // empty maps/objects
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/shared/support/support.ts]: the fix for the state being blank. - backend returns `{}` instead of `null` for the state when there was no override i guess

File: webapp/src/main/webapp/src/app/admin/tenant/component/property-override-tree-table/property-override-tree-table.component.ts (hunk 1 above)
Comment (ftdholbrook) [reply webapp/src/main/webapp/src/app/admin/tenant/component/property-override-tree-table/property-override-tree-table.component.ts]: Defaults (sourced from configprops) are masked with ****, if they come from the config server (get tenant for edit/view) they are encrypted.

File: webapp/src/main/webapp/src/assets/i18n/en.json [hunk 6]
```diff
@@ -1535,6 +1535,7 @@
     },
     "no-results": "There are no configuration properties matching your search",
     "boolean": {
+      "undefined": "False",
```
Comment (esotrope) [inline webapp/src/main/webapp/src/assets/i18n/en.json]: needed for switching tenants

File: webapp/src/main/webapp/src/app/admin/tenant/component/property-override-tree-table/property-override-tree-table.component.ts (hunk 1 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/admin/tenant/component/property-override-tree-table/property-override-tree-table.component.ts]: 👍
